src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.netstandard17.cs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Sockets
+{
+    public partial class TcpClient : IDisposable
+    {
+        // Initializes a new instance of the System.Net.Sockets.TcpClient class with the specified end point.
+        public TcpClient(IPEndPoint localEP)
+        {
+            if (NetEventSource.Log.IsEnabled())
+            {
+                NetEventSource.Enter(NetEventSource.ComponentType.Socket, this, nameof(TcpClient), localEP);
+            }
+
+            if (localEP == null)
+            {
+                throw new ArgumentNullException(nameof(localEP));
+            }
+
+            // IPv6: Establish address family before creating a socket
+            _family = localEP.AddressFamily;
+
+            InitializeClientSocket();
+            Client.Bind(localEP);
+
+            if (NetEventSource.Log.IsEnabled())
+            {
+                NetEventSource.Exit(NetEventSource.ComponentType.Socket, this, nameof(TcpClient), "");
+            }
+        }
+
+        // Initializes a new instance of the System.Net.Sockets.TcpClient class and connects to the specified port on 
+        // the specified host.
+        public TcpClient(string hostname, int port)
+        {
+            if (NetEventSource.Log.IsEnabled())
+            {
+                NetEventSource.Enter(NetEventSource.ComponentType.Socket, this, nameof(TcpClient), hostname);
+            }
+
+            if (hostname == null)
+            {
+                throw new ArgumentNullException(nameof(hostname));
+            }
+
+            if (!TcpValidationHelpers.ValidatePortNumber(port))
+            {
+                throw new ArgumentOutOfRangeException(nameof(port));
+            }
+            //
+            // IPv6: Delay creating the client socket until we have
+            //       performed DNS resolution and know which address
+            //       families we can use.
+            //
+            //InitializeClientSocket();
```

Comment:
The comment was from the desktop code; I think it was meant to convey that this call is what we're *not* doing here.  I've removed it.